### DISCUSSION TITLE
500 server error and error log in api/v1/widget/conversations? chatbox in website

### DISCUSSION THREAD

**Discussion by mdsami** [chatwoot/chatwoot#8632] [Self hosted]
![image](https://github.com/chatwoot/chatwoot/assets/5880703/1a628521-734a-4a63-88dc-260c9ae0e08c)

-- Logs begin at Sat 2023-12-30 07:44:01 UTC. --
Jan 03 07:11:46 ctwt chatwoot-web.1[384138]: [31ea3c1a-0fe7-4d59-8977-12336cf2e0cf] app/controllers/concerns/switch_locale.rb:11:in `switch_locale'
Jan 03 07:11:46 ctwt chatwoot-web.1[384138]: I, [2024-01-03T07:11:46.595979 #384138]  INFO -- : source=rack-timeout id=31ea3c1a-0fe7-4d59-8977-12336cf2e0cf timeout=15000ms service=122ms state=completed
Jan 03 07:11:46 ctwt chatwoot-web.1[384138]: I, [2024-01-03T07:11:46.790416 #384138]  INFO -- : source=rack-timeout id=765f9483-8a0a-4a57-89eb-7a8832dfba02 timeout=15000ms state=ready
Jan 03 07:11:46 ctwt chatwoot-web.1[384138]: I, [2024-01-03T07:11:46.791070 #384138]  INFO -- : [765f9483-8a0a-4a57-89eb-7a8832dfba02] Started GET "/api/v1/widget/conversations?website_token=FVDuVajQsrFs9PhTk8YUwASi&cw_conversation=eyJhbGciOiJIUzI1NiJ9.eyJzb3VyY2VfaWQiOiIwNmU2YTBlOC1kZmQ3LTRmYjAtOTAyOC05MWVmMWUyMDc3ZmUiLCJpbmJveF9pZCI6MX0.fh5hFgtAV01mxjhqP5Yr2TfzoxyCzcqHRiO1sIvJ6lg" for 172.68.231.155 at 2024-01-03 07:11:46 +0000
Jan 03 07:11:46 ctwt chatwoot-web.1[384138]: I, [2024-01-03T07:11:46.793017 #384138]  INFO -- : [765f9483-8a0a-4a57-89eb-7a8832dfba02] Processing by Api::V1::Widget::ConversationsController#index as JSON
Jan 03 07:11:46 ctwt chatwoot-web.1[384138]: I, [2024-01-03T07:11:46.793068 #384138]  INFO -- : [765f9483-8a0a-4a57-89eb-7a8832dfba02]   Parameters: {"website_token"=>"FVDuVajQsrFs9PhTk8YUwASi", "cw_conversation"=>"eyJhbGciOiJIUzI1NiJ9.eyJzb3VyY2VfaWQiOiIwNmU2YTBlOC1kZmQ3LTRmYjAtOTAyOC05MWVmMWUyMDc3ZmUiLCJpbmJveF9pZCI6MX0.fh5hFgtAV01mxjhqP5Yr2TfzoxyCzcqHRiO1sIvJ6lg"}
Jan 03 07:11:46 ctwt chatwoot-web.1[384138]: I, [2024-01-03T07:11:46.798994 #384138]  INFO -- : [765f9483-8a0a-4a57-89eb-7a8832dfba02]   Rendered api/v1/widget/conversations/index.json.jbuilder (Duration: 0.1ms | Allocations: 58)
Jan 03 07:11:46 ctwt chatwoot-web.1[384138]: I, [2024-01-03T07:11:46.799197 #384138]  INFO -- : [765f9483-8a0a-4a57-89eb-7a8832dfba02] Completed 200 OK in 6ms (Views: 0.4ms | ActiveRecord: 1.1ms | Allocations: 2435)
Jan 03 07:11:46 ctwt chatwoot-web.1[384138]: I, [2024-01-03T07:11:46.799591 #384138]  INFO -- : source=rack-timeout id=765f9483-8a0a-4a57-89eb-7a8832dfba02 timeout=15000ms service=9ms state=completed
Jan 03 07:11:59 ctwt chatwoot-web.1[384138]: I, [2024-01-03T07:11:59.647613 #384138]  INFO -- : RoomChannel#update_presence
Jan 03 07:12:15 ctwt chatwoot-web.1[384138]: I, [2024-01-03T07:12:15.384438 #384138]  INFO -- : source=rack-timeout id=6941556d-ced2-4eb9-9ad3-368df3ee990c timeout=15000ms state=ready
Jan 03 07:12:15 ctwt chatwoot-web.1[384138]: I, [2024-01-03T07:12:15.385087 #384138]  INFO -- : [6941556d-ced2-4eb9-9ad3-368df3ee990c] Started POST "/api/v1/widget/conversations?website_token=FVDuVajQsrFs9PhTk8YUwASi&cw_conversation=eyJhbGciOiJIUzI1NiJ9.eyJzb3VyY2VfaWQiOiIwNmU2YTBlOC1kZmQ3LTRmYjAtOTAyOC05MWVmMWUyMDc3ZmUiLCJpbmJveF9pZCI6MX0.fh5hFgtAV01mxjhqP5Yr2TfzoxyCzcqHRiO1sIvJ6lg&locale=en" for 172.68.231.155 at 2024-01-03 07:12:15 +0000
Jan 03 07:12:15 ctwt chatwoot-web.1[384138]: W, [2024-01-03T07:12:15.386238 #384138]  WARN -- : [6941556d-ced2-4eb9-9ad3-368df3ee990c] [Rack::Attack][Blocked] remote_ip: "172.68.231.155", path: "/api/v1/widget/conversations"
Jan 03 07:12:15 ctwt chatwoot-web.1[384138]: I, [2024-01-03T07:12:15.386376 #384138]  INFO -- : source=rack-timeout id=6941556d-ced2-4eb9-9ad3-368df3ee990c timeout=15000ms service=2ms state=completed
Jan 03 07:12:19 ctwt chatwoot-web.1[384138]: I, [2024-01-03T07:12:19.644323 #384138]  INFO -- : RoomChannel#update_presence

**Comment by vishnu-narayanan** [comment]
It looks like the `rack_attack` gem is throttling your IP. Could you try disabling `rack_attack` and see if it helps?

Try modifying the below environment variables

```
## Rack Attack configuration
## To prevent and throttle abusive requests
# ENABLE_RACK_ATTACK=true
# RACK_ATTACK_LIMIT=300
# ENABLE_RACK_ATTACK_WIDGET_API=true
```

references
1. https://www.chatwoot.com/docs/self-hosted/monitoring/rate-limiting/
2. https://github.com/chatwoot/chatwoot/blob/8a8f325f64c00caa8538e29a990ff4c671937436/.env.example#L212C1-L212C1
